test/unit/ngsiv2/HTTP_reveice_measures-test2.js

```diff
@@ -0,0 +1,391 @@
+                )
+                .query({ type: 'AnMQTTDevice' })
+                .reply(204);
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post(
+                    '/v2/entities/Second%20MQTT%20Device/attrs',
+                    utils.readExampleFile('./test/unit/ngsiv2/contextRequests/multipleMeasuresJsonTypes.json')
+                )
+                .query({ type: 'AnMQTTDevice' })
+                .reply(204);
+        });
+        it('should return a 200 OK with no error', function(done) {
+            request(optionsMeasure, function(error, result, body) {
+                should.not.exist(error);
+                result.statusCode.should.equal(200);
+                done();
+            });
+        });
+
+        it('should send its value to the Context Broker', function(done) {
+            request(optionsMeasure, function(error, result, body) {
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When a POST measure arrives with a TimeInstant attribute in the body', function() {
+        var optionsMeasure = {
+                url: 'http://localhost:' + config.http.port + '/iot/json',
+                method: 'POST',
+                json: [
+                    {
+                        humidity: '111222',
+                        TimeInstant: '20200222T222222'
+                    },
+                    {
+                        humidity: '111333',
+                        TimeInstant: '20200222T222222'
+                    }
+                ],
+                headers: {
+                    'fiware-service': 'smartGondor',
+                    'fiware-servicepath': '/gardens'
+                },
+                qs: {
+                    i: 'dev0130101',
+                    k: '1234'
+                }
+            },
+            provisionOptions = {
+                url: 'http://localhost:' + config.iota.server.port + '/iot/devices',
+                method: 'POST',
+                json: utils.readExampleFile('./test/deviceProvisioning/provisionDeviceTimeinstant.json'),
+                headers: {
+                    'fiware-service': 'smartGondor',
+                    'fiware-servicepath': '/gardens'
+                }
+            };
+
+        beforeEach(function(done) {
+            nock.cleanAll();
+            // This mock does not check the payload since the aim of the test is not to verify
+            // device provisioning functionality. Appropriate verification is done in tests under
+            // provisioning folder of iotagent-node-lib
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
```

Comment:
You should provide the expected payload for the /v2/entities?options=upsert operation here, similar to what you did in L152. Maybe you need to create a /test/unit/ngsiv2/contextRequests/timeInstantMeasures2.json file for it.